src/vs/workbench/contrib/remote/browser/remoteExplorer.ts

```diff
@@ -488,7 +488,15 @@ class ProcAutomaticPortForwarding extends Disposable {
 	) {
 		super();
 		this.notifier = new OnAutoForwardedAction(notificationService, remoteExplorerService, openerService, externalOpenerService, tunnelService, hostService, logService);
-		this._register(configurationService.onDidChangeConfiguration(async (e) => {
+		this.initialize();
+	}
+
+	private async initialize() {
+		if (!this.remoteExplorerService.tunnelModel.environmentTunnelsSet) {
+			await new Promise<void>(resolve => this.remoteExplorerService.tunnelModel.onEnvironmentTunnelsSet(() => resolve()));
```

Comment:
This await on environment tunnels being set was happening after all the listeners are set up. The listeners are meaningless until we have awaited the environment tunnels, so I moved it to happen earlier.